cmd/bnsd/app/app_performance_test.go

```diff
@@ -0,0 +1,134 @@
+package app
+
+import (
+	"encoding/hex"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/iov-one/weave/x/sigs"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+func BenchmarkBNSD(b *testing.B) {
+	var (
+		alice = weavetest.NewKey()
+		benny = weavetest.NewKey()
+		carol = weavetest.NewKey()
+		david = weavetest.NewKey()
+	)
+
+	type dict map[string]interface{}
+	genesis := dict{
+		"cash": []interface{}{
+			dict{
+				"address": alice.PublicKey().Condition().Address(),
+				"coins": []interface{}{
+					dict{
+						"whole":  123456789,
+						"ticker": "IOV",
+					},
+				},
+			},
+		},
+		"currencies": []interface{}{
+			dict{
+				"ticker": "IOV",
+				"name":   "Main token of this chain",
+			},
+		},
+		"distribution": []interface{}{
+			dict{
+				"admin": alice.PublicKey().Condition().Address(),
+				"recipients": []interface{}{
+					dict{"weight": 1, "address": benny.PublicKey().Condition().Address()},
+				},
+			},
+		},
+		"gconf": map[string]interface{}{
+			cash.GconfCollectorAddress: hex.EncodeToString(david.PublicKey().Condition().Address()),
+			cash.GconfMinimalFee:       coin.Coin{}, // no fee
+		},
+	}
+
+	cases := map[string]struct {
+		ops         func(weavetest.WeaveApp) error
+		wantChanged bool
+	}{
+		"empty block": {
+			ops: func(weavetest.WeaveApp) error {
+				// Without sleep this test is locking the CPU.
+				time.Sleep(time.Microsecond * 1000)
+				return nil
+			},
+			wantChanged: false,
+		},
+		"send coins from alice to carol": {
+			ops: func(wapp weavetest.WeaveApp) error {
+				tx := Tx{
+					Sum: &Tx_SendMsg{
+						&cash.SendMsg{
+							Src:    alice.PublicKey().Condition().Address(),
+							Dest:   carol.PublicKey().Condition().Address(),
+							Amount: coin.NewCoinp(0, 100, "IOV"),
+						},
+					},
+				}
+
+				const nonce = 0 // ?!?!?!?
+
+				sig, err := sigs.SignTx(alice, &tx, "mychain", nonce)
```

Comment:
I see your point about auto-generating addresses... the following lines could all collapse into one for 80% of the use-cases (a convenience method), and allow this full details when needed for other cases.